.github/workflows/nightly_snapshot.yaml

```diff
@@ -569,6 +569,9 @@ jobs:
                 https://api.github.com/repos/slint-ui/slint-cpp-templates-stm32/actions/workflows/ci.yaml/dispatches \
                 -d '{"ref":"main"}'
 
+    nightly-tests:
```

Comment:
The objective of 441695c130e52b2306bbc1f156fe2576d300cd1b was to move tests from the every-day run to run once a day. Could've put them into nightly_snapshot.yaml, but that's already bug. So that's why I created a separate .yaml file.

Up until this PR there's no dependency - it doesn't matter when the nightly tests are run. But with this PR, the esp-idf build of the nightly test will use the nightly binaries created in the depending job, hence this dependency.

Reply:
Could the dependency be reversed?  (nightly-test depends on nightly-snapshot)

Reply:
That's what this kind of implements, no?

```yaml
    nightly-tests:
        needs: [prepare_release]
```

nightly-tests depends on the bit of nightly-snapshot that's needed (not all of it).

Reply:
Ok, I can do an additional check in a follow-up.